packages/application-shell/src/components/application-shell/application-shell.js

```diff
@@ -368,23 +368,42 @@ RestrictedApplication.propTypes = {
   DEV_ONLY__loadNavbarMenuConfig: PropTypes.func,
 };
 
-const isBooleanAsString = environmentValue =>
-  environmentValue === 'true' || environmentValue === 'false';
+/**
+ * NOTE:
+ *   This function try-catches around a `JSON.parse` and return
+ *   the parsed value whenever possible.
+ *
+ *   This allows parsing most primitive values.
+ *
+ *   - `JSON.parse('null')` => `null`
+ *   - `JSON.parse('1')` => `1`
+ *   - `JSON.parse('["a", "b"]')` => `['a', 'b']`
+ */
+const getIfJSONValue = environmentValueAsString => {
+  let parsedEnvironmentValue;
+
+  try {
+    parsedEnvironmentValue = JSON.parse(environmentValueAsString);
+  } catch (e) {
+    return false;
+  }
+
+  return parsedEnvironmentValue;
+};
 
-const shallowlyCoerceBooleanValues = defaultMemoize(
-  uncoercedEnvironmentValues =>
-    Object.keys(uncoercedEnvironmentValues).reduce(
-      (coercedEnvironmentValues, key) => {
-        const uncoercedEnvironmentValue = uncoercedEnvironmentValues[key];
-        return {
-          ...coercedEnvironmentValues,
-          [key]: isBooleanAsString(uncoercedEnvironmentValue)
-            ? uncoercedEnvironmentValue === 'true'
-            : uncoercedEnvironmentValue,
-        };
-      },
-      {}
-    )
+const shallowlyCoerceValues = defaultMemoize(uncoercedEnvironmentValues =>
+  Object.keys(uncoercedEnvironmentValues).reduce(
+    (coercedEnvironmentValues, key) => {
+      const uncoercedEnvironmentValue = uncoercedEnvironmentValues[key];
+      return {
+        ...coercedEnvironmentValues,
+        [key]:
+          getIfJSONValue(uncoercedEnvironmentValue) ||
```

Comment:
Be careful, that if the actual parsed value is `false`, but the uncoerced value is `"false"`, then we will set a value of `"false"`. Same thing for e.g. number `0`.

So we don't need this `||`, we should just return the parsed value or not from the try-catch function. Do you know what I mean?

Reply:
Yeah you're right. Pushed the change.